### PR TITLE
Add PaywallData and its classes

### PR DESCRIPTION
**Pull request by vegaro**
Just adds the PaywallData class and its classes. Followup PRs will parse them and add them to the Offerings class

### CODE REVIEW

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallColor.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     * The underlying `Color`.
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    val underlyingColor: Color?,
+) {
+    /**
+     * Creates a color from a Hex string: `#RRGGBB` or `#RRGGBBAA`.
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    constructor(stringRepresentation: String) : this(
```
Comment (vegaro) [inline purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallColor.kt:25]: `valueOf` is only available from Android O +

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+            /**
+             * Image displayed as a background in a template.
+             */
+            val background: String? = null,
+
+            /**
+             * Image displayed as an app icon in a template.
+             */
+            val icon: String? = null,
```
Comment (vegaro) [inline purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:86]: `= null` is so we have the empty constructor

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+            /**
+             * Primary accent color.
+             */
+            val accent1: PaywallColor? = null,
+
+            /**
+             * Secondary accent color.
+             */
+            val accent2: PaywallColor? = null,
```
Comment (vegaro) [inline purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:135]: Same, these are  `= null` so there's a constructor without them. I saw the iOS code did this and thought it was convenient

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallColor.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package com.revenuecat.purchases.paywalls
+
+import android.graphics.Color
+import android.os.Build
+import androidx.annotation.RequiresApi
+
+/**
+ * Represents a color to be used by `RevenueCatUI`.
```
Comment (vegaro) [inline purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallColor.kt:8]: Left this as `RevenueCatUI` until we define the name of the library

**Review by tonidero — COMMENTED** [review]
This should be pointing to the `paywalls` branch I think, so we don't expose these before we have to. Nothing major, other than that 👍

**Comment by NachoSoto** [comment]
Nice!! I'll look at this more in detail tomorrow.
Should the PR target the `paywalls` branch?

**Comment by vegaro** [comment]
Updated to the `paywalls` branch

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt (hunk 2 above)
Comment (vegaro) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:86]: I believe that's the case yes https://github.com/RevenueCat/purchases-ios/blob/paywalls/Tests/UnitTests/Networking/Responses/Fixtures/Offerings.json#L75

Maybe @NachoSoto can confirm

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+    /**
+     * Generic configuration for any paywall.
+     */
+    val config: Configuration,
+
+    /**
+     * The base remote URL where assets for this paywall are stored.
+     */
+    val assetBaseURL: URL,
```
[the remark this answers is not in the document]
Comment (vegaro) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:25]: The iOS SDK also has this as a public property, and also builds the whole URL for each asset in a `TemplateViewConfiguration` that I believe is not public. I am not sure the reasoning why assetBaseURL is public though. cc: @NachoSoto

**Pull request merged**